server/network/aoprotocol.py

```diff
@@ -635,14 +635,15 @@ def net_cmd_ms(self, args):
                 "Your message is a repeat of the last one. Don't spam!")
             return
 
+        if evidence not in self.client.evi_list:
```

Comment:
Don't think this works the way you expect it to. The keyword `in` for an array checks the existence of a value, not a key.

Reply:
@oldmud0 I don't think this is working the way you think it does. evidence is an `int` and `self.client.evi_list` contains a list of ints.

This works properly! I swear it.I've attached an image showing the functionality of it. `evidence` is the int of the item that I am presenting. You can see that I used 0 for no evidence presented and 1 for when I presented an item from evidence.

![image](https://user-images.githubusercontent.com/36182383/106372000-3c3f5000-6339-11eb-82ad-f407eed48bc1.png)

Reply:
OHHH. Because the client's `evi_list` is actually a view of the area's evidence list in the case of HiddenCM mode. Gosh, that's so unintuitive.